workorders.lic

```diff
@@ -270,44 +270,75 @@ class WorkOrders
-    when 'What were'
-      stock_needed = 1
-      order_rem(info['stock-room'], stock_needed, info['stock-number'])
-    when 'You get'
-      /are (.*) parts left of the water/ =~ bput('count my water', 'are .* parts left of the water')
-      existing = Regexp.last_match(1).scan(/\w+/).to_a
-      val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'thirty' => 30, 'fourty' => 40, 'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
-      value = existing.map { |word| val_map[word] }.inject(&:+)
-      stock_needed = ((quantity * recipe['volume'] - value) / 100.0).ceil
+      when 'What were'
+        stock_needed = 1
+        order_rem(info['stock-room'], stock_needed, info['stock-number'])
+      when 'You get'
+        /are (.*) parts left of the water|is only (.*) part left/ =~ bput('count my water', 'are .* parts left of the water', 'is only .* part left')
+        water = Regexp.last_match(1).scan(/\w+/).to_a || Regexp.last_match(2).scan(/\w+/).to_a
+        val_map = { 'one' => 1, 'two' => 2, 'three' => 3, 'four' => 4, 'five' => 5, 'six' => 6, 'seven' => 7, 'eight' => 8, 'nine' => 9, 'ten' => 10, 'eleven' => 11, 'twelve' => 12, 'thirteen' => 13, 'fourteen' => 14, 'fifteen' => 15, 'sixteen' => 16, 'seventeen' => 17, 'eighteen' => 18, 'nineteen' => 19, 'twenty' => 20, 'twenty-one' => 21, 'twenty-two' => 22, 'twenty-three' => 23, 'twenty-four' => 24, 'twenty-five' => 25, 'twenty-six' => 26, 'twenty-seven' => 27, 'twenty-eight' => 28, 'twenty-nine' => 29, 'thirty' => 30, 'thirty-one' => 31, 'thirty-two' => 32, 'thirty-three' => 33, 'thirty-four' => 34, 'thirty-five' => 35, 'thirty-six' => 36, 'thirty-seven' => 37, 'thirty-eight' => 38, 'thirty-nine' => 39, 'forty' => 40, 'forty-one' => 41, 'forty-two' => 42, 'forty-three' => 43, 'forty-four' => 44, 'forty-five' => 45, 'forty-six' => 46, 'forty-seven' => 47, 'forty-eight' => 48, 'forty-nine' => 49,'fifty' => 50, 'sixty' => 60, 'seventy' => 70, 'eighty' => 80, 'ninety' => 90 }
+        value = water.map { |word| val_map[word] }.inject(&:+)
+        stock_needed = ((quantity * recipe['volume'] - value) / 100.0).ceil
     end
 
     if stock_needed < 3
       order_rem(info['stock-room'], stock_needed, info['stock-number'])
-      stow_tool('water')
+    end
+    case bput("get water from #{@bag}", "You get", "What were")
+      when 'You get'
+        bput('combine water', 'You combine')
+      when 'What were'
+        stow_tool('water')
     end
 
+    stow_tool('water')
     # order herbs
-
     order_rem(info['stock-room'], quantity, recipe['herb1_stock'])
-    if recipe['herb'] = !'na'
-      order_rem(info['stock-room'], stock_needed, recipe['herb2_stock'])
+     if recipe['herb2'] != 'na'
```

Comment:
Hello,

Okay, I can change to

if recipe['herb2'] != nil

and remove the empty herb2.

I am doing some more testing. :)

-- 
Regards,
Dr. Walter Wesley "Wes" Snyder V
484-300-4833 (h)
484-431-3355 (m)
484-302-7006 (f)

On 09/05/2016 04:00 PM, Robert Herbig wrote:
>
> In workorders.lic 
> <https://github.com/rpherbig/dr-scripts/pull/737#discussion_r77553371>:
>
> >      order_rem(info['stock-room'], quantity, recipe['herb1_stock'])
> > -    if recipe['herb'] = !'na'
> > -      order_rem(info['stock-room'], stock_needed, recipe['herb2_stock'])
> > +     if recipe['herb2'] != 'na'
>
> I'm not a fan of using 'na' to indicate herb2 is not applicable. 
> Omitting it from |base-crafting| would be easier, and then you could 
> check for nil here (which is more expressive in the code).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/rpherbig/dr-scripts/pull/737/files/bf67e4c8931441dbba649a92d114dc25d66e37ac#r77553371>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AU8XXE5t-dyjG5Gj5O7ec66uICBgLExuks5qnHTYgaJpZM4J0roH>.
>